openquakeplatform/openquakeplatform/updatures/app.py

```diff
@@ -0,0 +1,952 @@
+    new_pk       new key
+    """
+    md = MODELS_DESCR[model]
+    if md.pk_natural:
+        return False
+
+    if item['pk'] == new_pk:
+        return True
+
+    # search a previous defined item with the same pk
+    for item_same_pk in updates_gr[model]:
+        if item_same_pk['pk'] == new_pk:
+            old_pk = item['pk']
+            item['pk'] = maxpks[model]['maxpk'] + 1
+            update_pk(updates_gr, updatesk_gr, model, item_same_pk, maxpks,
+                      maxpks[model]['maxpk'])
+            item['pk'] = old_pk
+            break
+    else:
+        item_same_pk = None
+
+    # update all references
+    if md.natural is None:
+        for ref_model, ref_md in MODELS_DESCR.iteritems():
+            # found each model has a refs value associated with
+            # the current item model
+            for ref_reffield, ref in ref_md.refs.iteritems():
+                if isinstance(ref.model, types.FunctionType):
+                    continue
+
+                if ref.model != model:
+                    continue
+
+                updatures.UPD_LOG.log(0, "MDREF: %s" % ref_model)
+
+                for itemod in updates_gr[ref_model]:
+                    # if field not set or empty list continue
+                    ref_value = get_value(itemod, ref_reffield)
+                    if not ref_value:
+                        continue
+
+                    updatures.UPD_LOG.log(0, "ITEMOD: %s" % itemod)
+                    if ref.is_many:
+                        if type(ref_value[0]) is list:
+                            updatures.UPD_LOG.log(0, "itemod list of lists case not managed")
+                            sys.exit(10)
+                        for i, pk in enumerate(ref_value):
+                            if pk == item['pk']:
+                                ref_value[i] = new_pk
+                                break
+                    else:
+                        if ref_value == item['pk']:
+                            updatures.UPD_LOG.log(0, "UPDATE KEY: %s" % itemod)
+                            set_value(itemod, ref_reffield, new_pk)
+
+    # remove the item from the key based list of items
+    updatesk_gr[model].pop(key_get(md, item))
+
+    # update the pk with the new value
+    item['pk'] = new_pk
+
+    # add the item again to the key based list of items
+    updatesk_gr[model][key_get(md, item)] = item
+
+    ret = True
+    for backinhe in item.get('__backinhe__', []):
+        if ret:
+            update_pk(updates_gr, updatesk_gr, backinhe.model_descr.name,
+                      backinhe.item, maxpks, new_pk)
+
+    # if another item had the same pk value of new_pk before swap it
+    # with the old pk value of the updated item
+    if item_same_pk is not None:
+        if ret:
+            ret = update_pk(updates_gr, updatesk_gr, model, item_same_pk,
+                            maxpks, old_pk)
+
+    return ret
+
+def fk_compare(a, b):
+    b = list(b)
```

Comment:
It is unclear what you want to do in this function. A comment would be in order.